### PR TITLE
Extend eva to support legacy ConMon time/vert data

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
Extend `mon_data_space.py` to load legacy ConMon time/vert data files.  Additionally clean up `mon_data_space.py` by removing unused interface components and consolidate the "magic numbers" used in processing the legacy control files by moving them to the top of `class MonDataSpace` and giving them descriptive names.

Modify additional eva components to allow for batch processing of `DataType` in a manner like `Channel` and `Level`.  This is necessary to make the ConMon time plots.

Completes #167 

## Dependencies
 
None

## Impact

None

### CODE REVIEW

**Review by CoryMartin-NOAA — APPROVED** [review]
No issues that I can see!

**Comment by EdwardSafford-NOAA** [comment]
I found I'd missed some changes in handling station data when adding the DataType features.  I've also tacked on a change to support datasets for conmon vertical plots. The changes are small but it's probably worth another quick look @CoryMartin-NOAA .

File: src/eva/plotting/batch/base/plot_tools/figure_driver.py [hunk 1]
```diff
@@ -98,14 +98,21 @@ def figure_driver(config, data_collections, timing, logger):
             step_var_name = 'channel'
             title_fill = ' Ch. '
 
-            # Get list of levels, conditionally override step variables
+            # Get list of levels load step variables
```
Comment (CoryMartin-NOAA) [inline src/eva/plotting/batch/base/plot_tools/figure_driver.py]: might need a comma here?

**Review by CoryMartin-NOAA — APPROVED** [review]
Very minor thing but I think it's fine to merge with/without that comment edit, your call.

File: src/eva/plotting/batch/base/plot_tools/figure_driver.py (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply src/eva/plotting/batch/base/plot_tools/figure_driver.py]: Won't hurt to fix that.

**Pull request merged**